src/results-output/results-doc.ts

```diff
@@ -236,85 +236,147 @@ export function appendCurrentTopLevelForm() {
     appendFormGrabbingSessionAndNS(true);
 }
 
+function lastLineIsEmpty(doc: vscode.TextDocument): boolean {
+  const lines = doc.lineCount;
+  const lastLine = doc.getText(new vscode.Range(lines - 1, 0, lines - 1, Infinity));
+  return lastLine === "";
+}
+
 let scrollToBottomSub: vscode.Disposable;
 export interface OnAppendedCallback {
     (insertLocation: vscode.Location, newPosition?: vscode.Location): any
 }
-let editQueue: [string, OnAppendedCallback][] = [];
-let applyingEdit = false;
+
+type EditQueueItem = {
+  line: string ;
+  onAppended?: OnAppendedCallback;
+};
+type QueuedCallback = (item: EditQueueItem) => void;
+export class EditQueue {
+  #queue: EditQueueItem[] = [];
+
+  push(line: string, onAppended?: OnAppendedCallback) {
+    const item: EditQueueItem = {
+      line,
+      onAppended
+    }
+    this.#queue.push(item);
+    this.#queuedCallback?.(item);
+  }
+
+  #queuedCallback: QueuedCallback | undefined;
+
+  // Only supports a single callback.
+  onQueued(callback: QueuedCallback) {
+    this.#queuedCallback = callback;
+  }
+
+  nextBatch(maxBatchSize = 1000): EditQueueItem[] {
+    const itemBatch = takeWhile(this.#queue, (value, index) => {
+      return index < maxBatchSize;
+      // return index < maxBatchSize && !value[1];
+    });
+    const remainingEditQueue = this.#queue.slice(itemBatch.length);
+    this.#queue = remainingEditQueue;
+    return itemBatch;
+  }
+
+  size() {
+    return this.#queue.length;
+  }
+
+  clear() {
+    this.#queue = [];
+  }
+}
+
+const editQueue = new EditQueue();
+editQueue.onQueued(() => {
+  setTimeout(async () => {
+    while (editQueue.size() > 0) {
+      try {
+        await handleNextOutputQueueBatch();
+      } catch (err) {
+        console.error("Error writing output: ", err);
+      }
+    }
+  }, 100);
+});
```

Comment:
There's some room for improvement on how the queue gets flushed but this gets the job done.